CONTRIBUTING.md

```diff
@@ -0,0 +1,68 @@
+# Contributing to Hyperconverged Cluster Operator
+
+## ***This document is a work in progress***
+
+## Contributing to the HyperConverged API
+
+The Hyperconverged Cluster Operator represents an opinionated deployment of KubeVirt. Its purpose is to deploy KubeVirt
+and accompanying projects with good defaults, so they work together well for most people, in a testable and reproducible
+manner.
+
+This means that the API of HCO should be kept simple. HCO should do everything right out of the box, so it is easy to
+test and deploy. Sometimes, however, HCO cannot guess what is the right thing to do. On these rare occasions, a
+configurable is exposed in its Resource. Each configurable must be documented, so it is clear for a human operator when
+it should be used, and why the correct value cannot be guessed automatically.
+
+### Add new Feature Gate
+
+Think twice before you do. Feature gates make HCO very hard to test; each of them essentially duplicates our test
+matrix. They also complicate life for the human operator, who has to read the documentation to understand the
+implications of pressing a knob. You should add new featureGate only if hard-coding it to true considerably harms our
+typical users. Think if you can just add the new feature gate in a hard code manner to the requested operator.
+
+If there is a real need for a new feature gate, please follow these steps:
+
+1. In the PR message, describe the new feature gate, what it does and why it needed to be added to the HCO API.
+1. Add the new feature gate to the HyperConvergedFeatureGates struct
+   in [pkg/apis/hco/v1beta1/hyperconverged_types.go](pkg/apis/hco/v1beta1/hyperconverged_types.go)
+    - make sure the name of the feature gate field is as the feature gate field in the target operand, including casing.
+      It also must start with a capital letter, to be exposed from the api package.
+    - Set the field type to `FeatureGate`.
+    - Make sure the json name in the json tag is valid (e.g. starts with a small cap).
+    - add open-api annotations:
+        - add detailed description in the comment
+        - default annotation
+        - optional annotation
+
+for example:
+
+  ```golang
+  // Allow migrating a virtual machine with SRIOV interfaces.
+// When enabled virt-launcher pods of virtual machines with SRIOV
+// interfaces run with CAP_SYS_RESOURCE capability.
+// This may degrade virt-launcher security.
+// +optional
+// +kubebuilder:default=false
+SRIOVLiveMigration FeatureGate `json:"sriovLiveMigration,omitempty"`
```

Comment:
you mention this here; you most definitely need to add it to  docs/cluster-configuration.md so humans can know when they should use it, and why they don't want to have it by default.

Reply:
@dankenigsberg we're in the middle of reviewing all the feature gares now, and we'll handle all of them in a separate PR.

Also, this is an example of how to add a new feature gate, and not the addition itself.